api/V1/AuthenticateDevice.js

```diff
@@ -17,55 +17,31 @@ module.exports = function(app) {
   *
   * @apiSuccess {String} token JWT string to provide to further API requests
   */
-  app.post('/authenticate_device', function(req, res) {
+  app.post(
+    '/authenticate_device',
+    [
+      middleware.getDevice,
+      body('password').exists(),
+    ],
+    middleware.requestWrapper(async (request, response) => {
 
-    // Check that required data is given.
-    if (!req.body.devicename || !req.body.password) {
-      return responseUtil.send(res, {
-        statusCode: 400,
-        success: false,
-        messages: ['Missing devicename or devicename.']
-      });
-    }
-
-    models.Device.findOne({ where: { devicename: req.body.devicename } })
-      .then(function(device) {
-        // Return 401 if devicename is not found.
-        if (!device) {
-          return responseUtil.send(res, {
-            statusCode: 401,
-            success: false,
-            messages: ["No device found with given devicename"]
-          });
-        }
-
-        // Compare password.
-        device.comparePassword(req.body.password)
-          .then(function(passwordMatch) {
-            // Password is valid, send JWT in response.
-            if (passwordMatch) {
-              var data = device.getJwtDataValues();
-              var token = 'JWT ' + jwt.sign(data, config.server.passportSecret);
-              return responseUtil.send(res, {
-                statusCode: 200,
-                success: true,
-                messages: ["Successfull login."],
-                token: token
-              });
-            } else {
-              return responseUtil.send(res, {
-                statusCode: 401,
-                success: false,
-                messages: ["Wrong password or devicename."]
-              });
-            }
-          })
-          .catch(function(err) {
-            responseUtil.serverError(res, err);
-          });
-      })
-      .catch(function(err) {
-        responseUtil.serverError(res, err);
-      });
-  });
+      const passwordMatch = await request.body.device.comparePassword(request.body.password);
+      if (passwordMatch) {
+        var data = request.body.device.getJwtDataValues();
+        var token = 'JWT ' + jwt.sign(data, config.server.passportSecret);
+        return responseUtil.send(response, {
+          statusCode: 200,
+          success: true,
+          messages: ["Successfull login."],
```

Comment:
Successful

Reply:
Also, I think we were going to avoid the full stops at the end of messages?

Reply:
Yeah, I didn't go through changing all the messages on this commit. But will do that later.